analyzer/vulnerability-consumer/src/main/java/eu/fasten/analyzer/vulnerabilityconsumer/VulnerabilityConsumer.java

```diff
@@ -160,48 +160,65 @@ public long getMaxConsumeTimeout() {
+            if (context == null) {
+                throw new ContextNotFoundException();
+            }
             v.setFirstPatchedPurls(v.getFirstPatchedPurls().stream()
                     .filter(Objects::nonNull).collect(Collectors.toList()));
 
             var pkgIds = metadataUtility.getPackageIds(context, v);
-            if (pkgIds.size() == 0) return objectMapper.writeValueAsString(v);
+            if (pkgIds.size() == 0) {
+                throw new PackageNotFoundException();
+            }
 
             var pkgVersionIds = metadataUtility.getPackageVersionIds(v.getPurls(), context,
                     pkgIds, v);
             var pkgVersionPatchedIds = metadataUtility.getPackageVersionIds(v.getFirstPatchedPurls(),
                     context, pkgIds, null);
 
-            if (pkgVersionIds.size() == 0) return objectMapper.writeValueAsString(v);
+            if (pkgVersionIds.size() == 0) {
+                throw new PackageVersionsNotFoundException();
+            }
             var fastenUris = new HashSet<String>();
             var latestVersionId = pkgVersionIds.get(pkgVersionIds.size() - 1);
             pkgVersionPatchedIds.add(latestVersionId);
 
             checkFileExtensions(v);
 
             pkgVersionPatchedIds.forEach(pkgVersionId -> {
-               v.getPatches().forEach(p -> {
-                   logger.info("Searching for callables in " + p.getFileName() + " for Package Version ID: " + pkgVersionId);
-                   fastenUris.addAll(metadataUtility.getFastenUrisForPatch(p, pkgVersionId, context));
-               });
+                // TODO: Throw a custom exception here for package version IDs that do not have a patch at all.
+               if (v.getPatches().size() != 0) {
+                   v.getPatches().forEach(p -> {
+                       logger.info("Searching for callables in " + p.getFileName() + " for Package Version ID: " + pkgVersionId);
+                       fastenUris.addAll(metadataUtility.getFastenUrisForPatch(p, pkgVersionId, context));
+                   });
+                   writePatchDate(v);
+               } else {
+                   setPluginError(new PatchNotFoundException());
+               }
             });
 
+            // TODO: Throw a custom exception here for callables not found
             logger.info("Collected " + fastenUris.size() + " vulnerable fasten_uris ids");
-
             var fullFastenUris = new HashSet<String>();
             var vulnCallableIds = new HashSet<Long>();
             var vulnPkgVersionIds = new HashSet<>(pkgVersionIds);
-            fastenUris.forEach(uri -> {
-                var callIds = metadataUtility.getCallableIdsForFastenUri(uri, vulnPkgVersionIds, context);
-                callIds.forEach(id -> fullFastenUris.add(metadataUtility.getFullFastenUri(uri, id)));
-                vulnCallableIds.addAll(callIds);
-            });
 
-            v.setFastenUris(fullFastenUris);
-            writePatchDate(v);
```

Comment:
It seems that this call to `writePatchDate(v);` was removed. I'm not sure it was deliberate?